### ISSUE TITLE
[Text Selection] text selection should not have direct access to `TextEditingController`'s `TextEditingValue`

### ISSUE THREAD

**Issue by LongCatIsLooong** [flutter/flutter#79420]
Instead, it should operate on the text/selection from the previous frame as it is what the user sees.

**Comment by LongCatIsLooong** [comment]
The `TextSelectionDelegate` class is also used by `RenderEditable`. `RenderEditable` does not know the `TextEditingValue` used to build its `TextSpan`, or how to convert its `TextSpan` back to `TextEditingValue` (`_plainText` works for the default `TextEditingController.buildTextSpan` but if you want to do something less conventional, e.g., https://github.com/flutter/flutter/pull/80185, `_plainText` does not work). So either 
- `RenderEditable` needs to know what its `TextEditingValue` is, and how to build the text span, and converting `TextPosition`s from/to engine text offsets
- or non-idempotent APIs (delete, change the current selection, etc.) need to be removed from `RenderEditable`, or be changed to idempotent APIs (e.g.: `selectWord` => `getWordRangeAtPosition`), and let whoever knows how to do the `TextSpan` <=> `TextEditingValue` conversion handle the update.

**Comment by GaryQian** [comment]
I agree, RenderEditable does not need to know the raw TextEditingValue, especially since this value is changed by both the IME as well as various framework entities. It makes more sense to consolidate it in editable text where other similar operations are made.
